src/components/new-request/business-lookup.vue

```diff
@@ -0,0 +1,184 @@
+    <v-autocomplete
+      :hide-no-data="state != States.NO_RESULTS"
+      :items="searchResults"
+      :loading="state === States.SEARCHING"
+      :name="Math.random()"
+      :search-input.sync="searchField"
+      append-icon="mdi-magnify"
+      autocomplete="chrome-off"
+      autofocus
+      class="mt-5 mb-n2"
+      filled
+      hint="For example: &quot;Joe's Plumbing Inc.&quot;, &quot;BC1234567&quot;, &quot;FM1234567&quot;"
+      item-text="name"
+      item-value="identifier"
+      label="Business Name or Incorporation/Registration Number"
+      no-filter
+      persistent-hint
+      return-object
+      @input="onItemSelected($event)"
+      @keydown.enter.native.prevent
+    >
+      <template v-slot:append>
+        <v-progress-circular
+          v-if="state === States.SEARCHING"
+          color="primary"
+          indeterminate
+          :size="24"
+          :width="2"
+        />
+      </template>
+
+      <template v-slot:no-data>
+        <p class="pl-5 font-weight-bold">
+          No active B.C. business found
+        </p>
+        <p class="pl-5">
+          Ensure you have entered the correct business name or number.
+        </p>
+      </template>
+
+      <template v-slot:item="{ item }">
+        <v-row class="business-lookup-result pt-1">
+          <v-col cols="2" class="result-identifier">{{item.identifier}}</v-col>
+          <v-col cols="8" class="result-name">{{item.name}}</v-col>
+          <v-col cols="2" class="result-action">
+            <span v-if="item.disabled" class="added">Added</span>
+            <span v-else class="select">Select</span>
+          </v-col>
+        </v-row>
+      </template>
+    </v-autocomplete>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Emit, Watch } from 'vue-property-decorator'
+import { BusinessLookupResultIF } from '@/interfaces'
+import BusinessLookupServices from '@/services/business-lookup-services'
+import { Debounce } from 'vue-debounce-decorator'
+import Vue from 'vue'
+import { namespace } from 'vuex-class'
+
+const BusinessModule = namespace('business')
+
+enum States {
+  INITIAL = 'initial',
+  SEARCHING = 'searching',
+  SHOW_RESULTS = 'show results',
+  NO_RESULTS = 'no results',
+  SUMMARY = 'summary'
+}
+
+/*
+ * See PPR's BusinessSearchAutocomplete.vue for a Composition API example.
+ */
+@Component({})
+export default class BusinessLookup extends Vue {
+  // enum for template
+  readonly States = States
+
+  // action from business module
+  @BusinessModule.Action('isAffiliated')
+  private readonly isAffiliated!: (identifier: string) => Promise<boolean>
+
+  /** V-model for search field. */
+  searchField = ''
+
+  /** Results from business lookup API. */
+  searchResults = [] as Array<BusinessLookupResultIF>
+
+  /** State of this component. */
+  state = States.INITIAL
+
+  /** Called when searchField property has changed. */
+  @Watch('searchField')
+  @Debounce(600)
```

Comment:
Using the Debounce decorator here is OK for now since this is Namerequest, but eventually we will want to change it to the lodash debounce function or something similar. You don't have to change this now, but now you know about it.

Reply:
Kevin, please resolve this conversation when you have read it.

Reply:
I did